modules/graphdb/templates/00_functions.sh

```diff
@@ -11,6 +11,69 @@ log_with_timestamp() {
   echo "$(date '+%Y-%m-%d %H:%M:%S'): $1"
 }
 
+wait_for_vmss_nodes() {
+  local VMSS_NAME="$1"
+  local RESOURCE_GROUP="$2"
+  local RETRY_DELAY=10
+  local MAX_RETRIES=65
+  local RETRY_COUNT=0
+
+  # Get the desired capacity of the VMSS
+  local NODE_COUNT
+  NODE_COUNT=$(az vmss show \
+               --name "$VMSS_NAME" \
+               --resource-group "$RESOURCE_GROUP" \
+               --query "sku.capacity" \
+               --output tsv)
+
+  if ! [[ "$NODE_COUNT" =~ ^[0-9]+$ ]]; then
```

Comment:
`=` and `==` are for string comparisons and `-eq` is for numeric comparisons
Just use `-eq` and remove the regex